cvat/apps/engine/static/engine/js/shapes.js

```diff
@@ -1271,6 +1271,9 @@ class RaysModel extends PolyShapeModel {
     constructor(data, type, clientID, color) {
         super(data, type, clientID, color);
         this._vanishingPoint = data.vanishingPoint;
+        if (window.__UPDATE_RAYS__) {
+            console.log(this.vanishingPoint);
```

Comment:
remove debug log?

Reply:
It's intentional, I need to trigger getter logic here, sole `this.vanishingPoint;` on the line would look suspicious.

Reply:
You are abusing getter here then. Getters should not have side effects, because it is totally unclear from this code what the effect is. I.e. another developer will come next year, remove console.log and will spend days figuring out why things broke.

Reply:
I agree that side-effects in getters are not a good idea, but this code in `if (window.__UPDATE_RAYS__) { ... }` is just an ad-hoc to fix the data which were created in the first version of the tool. I'm going to delete this code later. Doing it the right way would require a bigger chunk of code.